modules/apps/29-fee/ibc_module.go

```diff
@@ -185,7 +185,7 @@ func (im IBCModule) OnRecvPacket(
 
 	forwardRelayer, found := im.keeper.GetCounterpartyAddress(ctx, relayer.String())
 	if !found {
-		forwardRelayer = ""
+		im.keeper.SetForwardRelayerAddress(ctx, channeltypes.NewPacketId(packet.GetSourceChannel(), packet.GetSourcePort(), packet.GetSequence()), forwardRelayer)
 	}
```

Comment:
We should be setting forward relayer address on recv packet, only if we haven't gotten an ACK and the counterparty address is found.

This is because we only need to save it if we aren't going to send it in the ack immediately